### ISSUE TITLE
Add app category display in Dev Center

### ISSUE THREAD

**Issue by jelveh** [HeyPuter/puter#1033]
Currently, the Dev Center doesn't display app categories in its interface. We should add category display both in the main app list and individual app settings pages to help developers better organize and identify their apps.

#### UI Changes Required

**Main App List:**
- Add a category label/badge next to each app name in the list
- Consider using subtle visual differentiation (like muted colors) to maintain readability
- Handle cases where category is undefined/empty gracefully (e.g., display "Uncategorized" or simply omit)
- Ensure proper alignment and spacing with existing app name and metadata

**Individual App Settings Page:**
- Add category field to the app's title or heading
- Real-time update of category display when edited

#### Technical Implementation

**Data Structure:**
- App categories are stored in the `metadata` attribute of app objects
- Example app object structure:
```javascript
{
  name: "my-app",
  metadata: {
    category: "productivity"
  },
  // ... other app properties
}
```

**Code Changes Required:**
1. Dev Center App List Component:
   - Update app list item rendering to include category
   - Add category styling/components
   - Handle category data fetching/display

2. App Update Logic:
   - Update local state/UI after successful category changes

#### UX Considerations
- Clear visual hierarchy: category should be visible but not dominate the app name
- Consistent styling across both list view and settings page
- Graceful handling of empty/undefined categories

#### Implementation Notes
- Files to modify are in `src/dev-center/`:

#### Testing Requirements
- Verify category display in app list
- Test category updates in settings
- Ensure proper handling of apps without categories
- Test UI responsiveness and layout
- Verify real-time updates after category changes

**Comment by Raghavendra-Aprameya** [comment]
Hi jelveh,

This is a great idea, and I’d love to work on it. I’ve reviewed the requirements and have some thoughts on how to approach the implementation, including adding category labels in the app list and handling updates on the settings page.

If no one else is currently assigned, may I take this up? Let me know if there are any specific guidelines or expectations you’d like me to follow.

Looking forward to your response!

Best regards,  
Aprameya Raghavendra.

**Comment by jelveh** [comment]
Thank you @Raghavendra-Aprameya, just assigned it to you. 

> Let me know if there are any specific guidelines or expectations you’d like me to follow.

So far only what's in the issue description. But if anything is vague, we could discuss it here.

Let me know if you need help with anything 🫡

**Comment by mariatouil** [comment]
@jelveh Hi,
I’ve submitted a Pull Request to address this issue: https://github.com/HeyPuter/puter/pull/1059.
Could you please review it and let me know if everything looks good?

Thank you in advance for your feedback!

**Comment by mariatouil** [comment]
@jelveh Hi,
I’ve submitted a Pull Request to address this issue: https://github.com/HeyPuter/puter/pull/1059.
Could you please review it and let me know if everything looks good?

Thank you in advance for your feedback!

**Comment by mariatouil** [comment]
@jelveh Hi,
I’ve submitted a Pull Request to address this issue: https://github.com/HeyPuter/puter/pull/1068.
Could you please review it and let me know if everything looks good?

Thank you in advance for your feedback!

**Comment by nikunjmathur08** [comment]
Hello @jelveh could you tell me what all is expected from this issue, like if you could provide me a brief about the design choice I have to make as I am sure you must be having a proper design theme, so could you give me some references so that I can solve this issue and help you guys out.

**Comment by ajaykrishnaswamy** [comment]
Can you assign this issue to me? I am part of the Headstarter community and would be happy to do this as I am pretty good with frontend.

**Comment by JoshJoseph1234** [comment]
Hi @jelveh I saw that in the main app list in dev center the app category is not shown. If this issue still remains could you assign it to me.

**Comment by jelveh** [comment]
Yes this is still open. Assigned! Let me know if you need help :)

**Comment by JoshJoseph1234** [comment]
Hi @jelveh just completed and i have created a PR to this issue #1033. Let me know if any changes are needed
